apollo-router/src/plugins/telemetry/endpoint.rs

```diff
@@ -20,8 +20,8 @@ pub(crate) struct UriEndpoint {
 
 impl UriEndpoint {
     /// Converts an endpoint to a URI using the default endpoint as reference for any URI parts that are missing.
-    pub(crate) fn to_uri(&self, default_endpoint: &Uri) -> Option<Uri> {
-        self.uri.as_ref().map(|uri| {
+    pub(crate) fn extend_to_full_uri(&self, default_endpoint: &Uri) -> Uri {
```

Comment:
I tried to find a better name for something that this function is doing:
> Converts an endpoint to a URI using the default endpoint as reference for any URI parts that are missing.

Reply:
i think `to_uri` still makes sense as a name. it's a conversion, and [mostly] follows rust's [naming conventions](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). the only thing that's different is that we do pass in a default value. `extend_to_full_uri` seems more confusing to me.

Reply:
@lrlna just to clarify, this function extends stuff based on default endpoint.
For example, you can create `Uri` without a port, but this function ensures that port is always copied from default.

For me this is surprising behaviour, since I expect `to_uri` just convert stuff to `Uri` but in reality it returns "full Uri" by extending Uri with parts from default.

Reply:
maybe just `to_full_uri` . Usually in rust std lib when it's an `extend_*` method it takes `&mut self` instead. So either you change the prototype to take `&mut self` and keep `extend` or the other solution

Reply:
I like `to_full_uri` and agree that `extend_` is confusing 👍

Reply:
`to_full_uri` wfm